### PR TITLE
Reland the fix for fuzzer stats showing /coverage-report links (#145).

### PR DESCRIPTION
**Pull request by Dor1s**
This relands Abhishek's #103 and fixes the tests. The idea is that we should not use `/coverage-report` links on the fuzzer stats page, as such links may not be valid. We use those links only on OSS-Fuzz main page, as we don't want to query data from datastore for that. On fuzzer stats page, we query the data and can certainly know whether we have a corresponding coverage report or not. If we do have it, we put a valid link, otherwise `--`.

### CODE REVIEW

**Comment by Dor1s** [comment]
My current error is:

```
ERROR: test_build_by_job (tests.appengine.handlers.fuzzer_stats_test.TestBuildResults)
Tests basic build_results with valid parameters (group by job).
----------------------------------------------------------------------
Traceback (most recent call last):
  File "<redacted>/clusterfuzz/src/python/tests/appengine/handlers/fuzzer_stats_test.py", line 197, in test_build_by_job
    result = fuzzer_stats.build_results(*build_args)
  File "<redacted>/clusterfuzz/src/appengine/handlers/fuzzer_stats.py", line 302, in build_results
    return _build_old_results(fuzzer, jobs, group_by, date_start, date_end)
  File "<redacted>/clusterfuzz/src/python/base/memoize.py", line 207, in wrapper
    result = func(*args, **kwargs)
  File "<redacted>/clusterfuzz/src/appengine/handlers/fuzzer_stats.py", line 321, in _build_old_results
    return _build_results(fuzzer, jobs, group_by, date_start, date_end)
  File "<redacted>/clusterfuzz/src/appengine/handlers/fuzzer_stats.py", line 368, in _build_results
    _build_rows(result, columns, results['rows'], group_by)
  File "<redacted>/clusterfuzz/src/appengine/handlers/fuzzer_stats.py", line 256, in _build_rows
    data = column.field.get(group_by, first_column_value)
  File "<redacted>/clusterfuzz/src/python/metrics/fuzzer_stats.py", line 605, in get
    logs_path = self._get_logs_bucket_path(group_by, group_by_value)
  File "<redacted>/clusterfuzz/src/python/metrics/fuzzer_stats.py", line 596, in _get_logs_bucket_path
    logs_bucket = self.ctx.get_logs_bucket(fuzzer_name=fuzzer, job_type=job)
  File "<redacted>/clusterfuzz/src/python/metrics/fuzzer_stats.py", line 559, in get_logs_bucket
    return self._get_logs_bucket_from_job(job_type)
  File "<redacted>/clusterfuzz/src/python/base/memoize.py", line 207, in wrapper
    result = func(*args, **kwargs)
  File "<redacted>/clusterfuzz/src/python/metrics/fuzzer_stats.py", line 537, in _get_logs_bucket_from_job
    'FUZZ_LOGS_BUCKET')
  File "<redacted>/clusterfuzz/src/python/datastore/data_handler.py", line 964, in get_value_from_job_definition
    job = data_types.Job.query(data_types.Job.name == job_type).get()
  File "/usr/local/google/home/mmoroz/google-cloud-sdk/platform/google_appengine/google/appengine/ext/ndb/model.py", line 999, in __eq__
    return self._comparison('=', value)
  File "/usr/local/google/home/mmoroz/google-cloud-sdk/platform/google_appengine/google/appengine/ext/ndb/model.py", line 987, in _comparison
    value = self._do_validate(value)
  File "/usr/local/google/home/mmoroz/google-cloud-sdk/platform/google_appengine/google/appengine/ext/ndb/model.py", line 1092, in _do_validate
    value = self._call_shallow_validation(value)
  File "/usr/local/google/home/mmoroz/google-cloud-sdk/platform/google_appengine/google/appengine/ext/ndb/model.py", line 1284, in _call_shallow_validation
    return call(value)
  File "/usr/local/google/home/mmoroz/google-cloud-sdk/platform/google_appengine/google/appengine/ext/ndb/model.py", line 1331, in call
    newvalue = method(self, value)
  File "/usr/local/google/home/mmoroz/google-cloud-sdk/platform/google_appengine/google/appengine/ext/ndb/model.py", line 1781, in _validate
    (value,))
BadValueError: Expected string, got 20447464223
```

**Comment by Dor1s** [comment]
PTAL, should be good now. I'll re-test on staging before deploying (I think we tested the #103 patch on both staging servers, but it never hurts to verify once again).

File: src/python/metrics/fuzzer_stats.py [hunk 1]
```diff
@@ -385,14 +384,18 @@ def get_coverage_info(self, group_by, group_by_value):
...
       coverage_info = self.ctx.get_coverage_info(self.ctx.fuzzer,
                                                  group_by_value)
 
     elif group_by == QueryGroupBy.GROUP_BY_FUZZER:
-      # Return latest coverage data.
+      # Return latest coverage data for each fuzzer.
       coverage_info = self.ctx.get_coverage_info(group_by_value)
 
+    elif group_by == QueryGroupBy.GROUP_BY_JOB:
+      # Return the latest coverage data for the fuzzer.
```
Comment (oliverchang) [inline src/python/metrics/fuzzer_stats.py]: could you add a comment here explaining why the condition here is GROUP_BY_JOB but the code says self.ctx.fuzzer?

Comment (Dor1s) [reply src/python/metrics/fuzzer_stats.py]: Good point, done!

**Comment by Dor1s** [comment]
Seems to work, though I've noticed that we do not respect the date when group by fuzzer or by job and always return the latest coverage information. Will file a separate bug for that, as it's not critical, plus I suspect that it might be an issue for some other stats columns.

**Pull request merged**